bokehjs/src/coffee/models/tickers/log_ticker.coffee

```diff
@@ -31,32 +31,31 @@ export class LogTicker extends AdaptiveTicker
         factors = range(start_factor, end_factor + 1)
 
       ticks = (factor * interval for factor in factors when factor != 0)
+      ticks = ticks.filter((tick) -> data_low <= tick <= data_high)
 
-      if num_minor_ticks > 1
+      if num_minor_ticks > 0 and ticks.length > 0
         minor_interval = interval / num_minor_ticks
-        minor_offsets = (i*minor_interval for i in [1..num_minor_ticks])
-        for x in minor_offsets
+        minor_offsets = (i*minor_interval for i in [0...num_minor_ticks])
+        for x in minor_offsets[1..minor_offsets.length]
           minor_ticks.push(ticks[0]-x)
         for tick in ticks
           for x in minor_offsets
             minor_ticks.push(tick+x)
     else
-      startlog = Math.ceil(log_low)
-      endlog = Math.floor(log_high)
+      startlog = Math.ceil(log_low * 0.9999)
```

Comment:
It doesn't always miss the tick, only when there's enough floating point error introduced in ``log_low = Math.log(data_low) / Math.log(base)``. And it's not limited to the right hand side.

We would incorrectly add a tick if the range is supposed to be 2.9999997 (for example), but I'm unsure how common of a case or how undesirable a 3.0 tick is in that case.